app/code/VConnect/OrderVolume/Setup/Patch/Data/AddProductEntityVolumeAttribute.php

```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types=1);
+
+namespace VConnect\OrderVolume\Setup\Patch\Data;
+
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Eav\Setup\EavSetup;
+use Magento\Eav\Setup\EavSetupFactory;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Eav\Model\Entity\Attribute\ScopedAttributeInterface;
+
+class AddProductEntityVolumeAttribute implements DataPatchInterface
+{
+    public function __construct(
+        private ModuleDataSetupInterface $moduleDataSetup,
+        private EavSetupFactory $eavSetupFactory
+    ) {}
+
+
+    public function apply()
+    {
+        /** @var EavSetup $eavSetup */
+        $eavSetup = $this->eavSetupFactory->create(['setup' => $this->moduleDataSetup]);
+        $eavSetup->addAttribute(
+            Product::ENTITY,
+            'volume',
+            [
+                'group' => 'General',
+                'type' => 'decimal',
+                'label' => 'Product Volume',
+                'input' => 'text',
+                'required' => true,
```

Comment:
probably false. virtual products doesn't have weight and volume at all. Example: virtual giftcard.